requests/sessions.py

```diff
@@ -157,13 +157,11 @@ def resolve_redirects(self, response, stream=False, timeout=None,
             if response.is_permanent_redirect and request.url != prepared_request.url:
                 self.redirect_cache[request.url] = prepared_request.url
 
-            old_method = prepared_request.method
-            self.rebuild_method(prepared_request, response)
-            new_method = prepared_request.method
+            method_changed = self.rebuild_method(prepared_request, response)
```

Comment:
Eh, so I guess I'm indifferent on this now that I'm really looking at it. Pulling the logic from the conditional into it's own function adds code without significantly reducing the `resolve_redirects` footprint.

The return value for `rebuild_method` provides an op vs no-op check but isn't particularly useful. I threw it in there as a slight readability improvement, but we can ditch it if you aren't feeling it.